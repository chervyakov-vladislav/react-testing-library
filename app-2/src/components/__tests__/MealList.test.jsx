import { screen } from '@testing-library/react';

import { renderWithRouter } from '../../testUtils/routerProvider';
import { MealList } from '../MealList';

describe('MealList', () => { 
  it('рендер компонента', () => {
    const listProps = [
      {
        strMeal: 'Cheese',
        idMeal: '1',
        strMealThumb: '/meal.png',
      },
      {
        strMeal: 'meat',
        idMeal: '2',
        strMealThumb: '/meal.png',
      }
    ] 

    renderWithRouter(<MealList meals={listProps}/>);

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});