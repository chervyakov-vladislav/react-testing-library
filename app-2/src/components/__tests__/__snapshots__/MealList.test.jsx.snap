// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealList рендер компонента 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Cheese"
        src="/meal.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Cheese
      </span>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/meal/1"
      >
        Watch recipe
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="meat"
        src="/meal.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        meat
      </span>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/meal/2"
      >
        Watch recipe
      </a>
    </div>
  </div>
</div>
`;
